jsonnet/components/kube-state-metrics.libsonnet

```diff
@@ -307,4 +307,48 @@ function(params)
     },
 
     customResourceStateConfigmap: crsConfig,
+    
+    // This networkpolicy allow access to kube-state-metrics port 8443
+    networkPolicy: {
+      apiVersion: 'networking.k8s.io/v1',
+      kind: 'NetworkPolicy',
+      metadata: {
+        annotations: {
+          'include.release.openshift.io/hypershift': 'true',
+          'include.release.openshift.io/ibm-cloud-managed': 'true',
+          'include.release.openshift.io/self-managed-high-availability': 'true',
+          'include.release.openshift.io/single-node-developer': 'true',
+        },
+        name: 'kube-state-metrics-access',
+        namespace: cfg.namespace,
+      },
+      spec: {
+        podSelector: {
+          matchLabels: {
+            'app.kubernetes.io/name': 'kube-state-metrics',
+          },
+        },
+        policyTypes: [
+          'Ingress',
+          'Egress',
+        ],
+        ingress: [
+          {
+            ports: [
+              {
+                port: '8443',
+                protocol: 'TCP',
+              },
+              {
+                port: '9443',
+                protocol: 'TCP',
+              },
+            ],
+          },
+        ],
+        egress: [
```

Comment:
Is there any recommendation to allow egress traffic only to the API service? This is the only destination which ksm should try to connect but I guess that it's not trivial to accommodate classic OCP (self-managed) and hosted clusters (HyperShift).

Reply:
networkpolicy applied to pod only, seems we can not define for API service, will check again.